### PR TITLE
Heart fixes / tweaks

### PR DESCRIPTION
**Pull request by Vivalas**
Heart death will now actually trigger heart attacks.

Additionally, hearts now adjust blood_volume calculations by up to -50% blood at max damage. Past 30 heart damage you now get fluff messages about your heart beating real fast, your heart beats real fast, and you get external indicators of heart damage such as panting and brief stuns.

:cl: Vivalas
tweak: Heart damage is now actually somewhat relevant. Effective blood volume for oxyloss calculations now scales linearly with heart damage, at max -50%.
fix: Heart death now properly triggers heart attacks
/:cl:

### CODE REVIEW

File: code/modules/surgery/organs/organ_internal.dm [hunk 1]
```diff
@@ -174,13 +183,17 @@
 	Restart()
 	..()
 
+/obj/item/organ/internal/heart/necrotize()
+	..()
+	beating = FALSE
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ_internal.dm]: Use the proper proc for setting a heart attack.

File: code/modules/surgery/organs/blood.dm [hunk 2]
```diff
@@ -86,6 +86,10 @@
 
 		//Effects of bloodloss
 		var/oxy_immune = species.flags & NO_BREATHE //Some species have blood, but don't breathe; they should still suffer the effects of bloodloss.
+		if(can_heartattack()) //if we have no need for a heart it doesn't affect blood volume
+			var/obj/item/organ/internal/heart/heart = get_int_organ(/obj/item/organ/internal/heart)
+			if(heart && heart.beating) //Let's not make heart attacks worse...
+				blood_volume = max(0,round(blood_volume - blood_volume * (heart.damage / (heart.max_damage * 2)))) //up tp -50% blood_volume for a totalled heart
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/blood.dm]: I dunno why you didn't just make this: 

`blood_volume *= (1 - (heart.damage / (heart.max_damage * 2)))`

Comment (Vivalas) [inline code/modules/surgery/organs/blood.dm]: I cannot into math

**Comment by Fox-McCloud** [comment]
Hear stopping upon death of the organ makes a lot of sense, and I'm all for that.

That said, I'm against heart damage inherently decreasing blood volume; we had this in the past; needless to say, it helps make the already egregiously overpowered brute damage all that much worse; not only is it causing bleeding, potentially internal bleeding, embedding, broken bones, but now it's causing bonus oxygen damage.

This is particularly bad for projectiles; usually it only took a round or two (even if it was moderate damage) to take someone out of the round (this happened with wizards a *lot*).

File: code/modules/surgery/organs/organ_internal.dm [hunk 3]
```diff
@@ -185,7 +185,7 @@
 
 /obj/item/organ/internal/heart/necrotize()
 	..()
-	beating = FALSE
+	owner.set_heartattack(TRUE)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ_internal.dm]: Need to check for owner, first; the heart can necrotize out of the body.

**Comment by Vivalas** [comment]
Currently the oxyloss is minimal (caps at around 5) until heart damage is at 45, at which point it seems to slowly climb until you collapse and die. It takes around 3 minutes to go. Changing it to -33% (heart.max_damage * 3) makes this happen at 50 damage but I think that's too generous, given you're about to have a heart attack anyways at that point.

File: code/modules/surgery/organs/organ_internal.dm [hunk 4]
```diff
@@ -174,13 +183,18 @@
 	Restart()
 	..()
 
+/obj/item/organ/internal/heart/necrotize()
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ_internal.dm]: For now, this is the only thing we'd like included in this PR, from a balance altering standpoint.

**Comment by Fox-McCloud** [comment]
Stale; please update or this will be closed.

File: code/modules/surgery/organs/organ_internal.dm (hunk 4 above)
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ_internal.dm]: As I commented earlier, this should be the only thing in this PR, aside from the `TRUE` and `FALSE` changes.

It's why this hasn't been merged yet.

**Comment by Vivalas** [comment]
Just a note, will re open this for a proper review once I fix conflicts.

**Comment by Fox-McCloud** [comment]
There was an alternative PR merged taht addressed this.

**Pull request closed without merging**